eventmesh-connectors/eventmesh-connector-http/src/main/java/org/apache/eventmesh/connector/http/sink/handle/WebhookHttpSinkHandler.java

```diff
@@ -0,0 +1,211 @@
+import org.apache.eventmesh.connector.http.util.HttpUtils;
+import org.apache.eventmesh.openconnect.offsetmgmt.api.data.ConnectRecord;
+
+import java.net.URI;
+import java.util.concurrent.BlockingQueue;
+import java.util.concurrent.LinkedBlockingQueue;
+import java.util.concurrent.TimeUnit;
+
+import io.netty.handler.codec.http.HttpResponseStatus;
+import io.vertx.core.Future;
+import io.vertx.core.Vertx;
+import io.vertx.core.buffer.Buffer;
+import io.vertx.core.http.HttpHeaders;
+import io.vertx.core.http.HttpMethod;
+import io.vertx.core.http.HttpServer;
+import io.vertx.core.http.HttpServerOptions;
+import io.vertx.ext.web.Router;
+import io.vertx.ext.web.client.HttpResponse;
+import io.vertx.ext.web.handler.LoggerHandler;
+
+import com.alibaba.fastjson2.JSONObject;
+
+import lombok.extern.slf4j.Slf4j;
+
+/**
+ * Extends CommonHttpSinkHandler to provide additional functionality for handling webhook features, including sending requests to callback servers,
+ * allowing longer response wait times, storing responses returned from callback servers, and exposing received data through an HTTP service.
+ */
+@Slf4j
+public class WebhookHttpSinkHandler extends CommonHttpSinkHandler {
+
+    // the configuration for webhook
+    private final HttpWebhookConfig webhookConfig;
+
+    // the server for exporting the received data
+    private HttpServer exportServer;
+
+    // store the received data, when webhook is enabled
+    private final BlockingQueue<Object> receivedDataQueue;
+
+    public WebhookHttpSinkHandler(SinkConnectorConfig sinkConnectorConfig) {
+        super(sinkConnectorConfig);
+        this.webhookConfig = sinkConnectorConfig.getWebhookConfig();
+        this.receivedDataQueue = new LinkedBlockingQueue<>();
+        // init the export server
+        doInitExportServer();
+    }
+
+    /**
+     * Initialize the server for exporting the received data
+     */
+    private void doInitExportServer() {
+        final Vertx vertx = Vertx.vertx();
+        final Router router = Router.router(vertx);
+        // add logger handler
+        router.route().handler(LoggerHandler.create());
+        // add export handler
+        router.route()
+            .path(this.webhookConfig.getExportPath())
+            .method(HttpMethod.GET)
+            .produces("application/json")
+            .handler(ctx -> {
+                // get received data
+                Object data = this.receivedDataQueue.poll();
+                if (data != null) {
+
+                    // export the received data
+                    ctx.response()
+                        .putHeader(HttpHeaders.CONTENT_TYPE, "application/json; charset=utf-8")
+                        .setStatusCode(HttpResponseStatus.OK.code())
+                        .send(JSONObject.of("data", data).toJSONString());
+                    if (log.isDebugEnabled()) {
+                        log.debug("Succeed to export callback data. Data: {}", data);
```

Comment:
How about returning multiple data in batches at once? That is, pagination, which can reduce the number of times the endpoint is called.

It is necessary to remind users in the `/export` document that the request to the endpoint is not idempotent, and the data stored on the server will be deleted after the request. This may be a bit inconvenient. It would be better if there is a GET path parameter, the default value of which is not to delete data.

Reply:
We might need to make some changes as using a queue structure may not fulfill this requirement. 
Possible parameters could include: `pageNum=1`, `pageSize=10`, `type=poll/peek`. 

Do you have any suggestions?

Reply:
> Possible parameters could include: `pageNum=1`, `pageSize=10`, `type=poll/peek`.

These are enough~

>We might need to make some changes as using a queue structure may not fulfill this requirement. 

A temp array may need to be extracted.

Reply:
I plan to achieve the above functionality by adding the following code:
```java
public List<Object> getPageItems(int pageNum, int pageSize, boolean removed) {
        int startIndex = (pageNum - 1) * pageSize;
        int endIndex = startIndex + pageSize;
        Iterator<Object> iterator = receivedDataQueue.iterator();

        List<Object> pageItems = new ArrayList<>(pageSize);
        int count = 0;
        while (iterator.hasNext() && count < endIndex) {
            Object item = iterator.next();
            if (count >= startIndex) {
                pageItems.add(item);
                if (removed) {
                    iterator.remove();
                }
            }
            count++;
        }
        return pageItems;
    }
```
But there are some issues with the above approach:

1. It iterates from the beginning every time.
2. If using a `LinkedBlockingQueue`, there are thread safety concerns, requiring the use of the `synchronized` or other mechanisms to ensure thread safety. Alternatively, using `ConcurrentLinkedQueue` as a replacement is an option, but it has slightly worse performance for removing elements from the middle. (I might conduct some tests and choose one of these solutions).

Reply:
>It iterates from the beginning every time.

Acceptable for a 5k size queue.

>using ConcurrentLinkedQueue as a replacement is an option, but it has slightly worse performance for removing elements from the middle.

If the user wants to poll (i.e. delete data from the queue), the value of pageNum should always be 1, which ensures the order and integrity of the queue data.

Reply:
> If the user wants to poll (i.e. delete data from the queue), the value of pageNum should always be 1, which ensures the order and integrity of the queue data.

If we have this precondition, then I think we can use `ConcurrentLinkedQueue` to replace `LinkedBlockingQueue`.

Reply:
Resolved at https://github.com/apache/eventmesh/pull/4837/commits/9c9e4a85706e42e5da00381ac4447644ca6f4d14